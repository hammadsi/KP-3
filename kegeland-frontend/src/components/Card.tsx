import { Box, Skeleton } from '@chakra-ui/react';
import { ComponentProps } from 'react';

type CardProps = {
  loading?: boolean;
  containerProps?: ComponentProps<typeof Skeleton>;
} & ComponentProps<typeof Box>;

const Card: React.FC<CardProps> = ({
  loading,
  children,
  containerProps,
  ...props
}) => {
  return (
    <Box
      padding={4}
      backgroundColor="white"
      shadow="md"
      justify="center"
      align="center"
      borderRadius={5}
      marginBottom={5}
      {...props}>
      <Skeleton isLoaded={!loading} {...containerProps} h="100%" w="100%">
        {children}
      </Skeleton>
    </Box>
  );
};

export default Card;
