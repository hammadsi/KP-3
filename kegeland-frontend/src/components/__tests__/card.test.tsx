import renderer from 'react-test-renderer';

import Card from '../Card';

describe('Test card', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card loading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
