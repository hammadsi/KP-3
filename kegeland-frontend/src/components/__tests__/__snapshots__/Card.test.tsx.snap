// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test card renders correctly 1`] = `
<div
  align="center"
  className="css-190jc56"
  justify="center"
>
  <div
    className="chakra-skeleton css-1i0luk8"
  />
</div>
`;
